scipion/install/funcs.py

```diff
@@ -296,30 +296,31 @@ def getPythonPackagesFolder():
 
     @staticmethod
     def getIncludeFolder():
-        return '%s/include' % (Environment.getSoftware())
+        return Environment.getSoftware('include')
 
     def getLib(self, name):
-        return '%s/lib%s.%s' % (Environment.getLibFolder(),
-                                name, self._libSuffix)
+
+        return Environment.getLibFolder('lib%s.%s' % (name, self._libSuffix))
 
     @staticmethod
-    def getBinFolder():
-        return '%s/bin' % Environment.getSoftware()
+    def getBinFolder(*paths):
+        return  os.path.join(mkdir(Environment.getSoftware('bin')), *paths)
 
     @staticmethod
     def getBin(name):
-        return '%s/%s' % (Environment.getBinFolder(), name)
+        return  Environment.getBinFolder(name)
 
     @staticmethod
     def getTmpFolder():
-        return '%s/tmp' % Environment.getSoftware()
+        return  mkdir(Environment.getSoftware('tmp'))
+
+    @staticmethod
+    def getLogFolder(*path):
+        return os.path.join(mkdir(Environment.getSoftware('log')),*path)
 
     @staticmethod
     def getEmFolder():
-        # Create it if it does not exists
-        if not exists(pwem.Config.EM_ROOT):
-            os.makedirs(pwem.Config.EM_ROOT)
-        return pwem.Config.EM_ROOT
+        return mkdir(pwem.Config.EM_ROOT)
```

Comment:
Same here, mkdir command everytime?

Reply:
Still, it checks if the folder exists every time you call the function, anyway I think these functions are not called too much

Reply:
tmp folder was created with ./scipion config .... and is required for installation. Now it is needed. Maybe now it can be done in the installation process but I think, it is the same. Then if it is already created it does nothing

Reply:
It is not the same that you explicitly create it (when needed) than every time you call a get function, it tries to create it (or check if exists)

Reply:
Maybe we can trigger some kind of "folder creation" when calling some installation code?